core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -228,4 +232,29 @@ object FileUtils {
   } catch (e: IOException) {
     "".also { e.printStackTrace() }
   }
+
+  @SuppressLint("WrongConstant")
+  @JvmStatic fun getPathFromUri(activity: Activity, data: Intent): String {
+    val uri: Uri? = data.data
+    val takeFlags: Int = data.flags and (Intent.FLAG_GRANT_READ_URI_PERMISSION
+      or Intent.FLAG_GRANT_WRITE_URI_PERMISSION)
+    activity.grantUriPermission(
+      activity.packageName, uri,
+      Intent.FLAG_GRANT_READ_URI_PERMISSION or Intent.FLAG_GRANT_WRITE_URI_PERMISSION
+    )
+    activity.contentResolver.takePersistableUriPermission(uri!!, takeFlags)
+
+    val dFile = DocumentFile.fromTreeUri(activity, uri)
+    val originalPath = dFile!!.uri.path!!.substring(
+      dFile.uri.path!!.lastIndexOf(":") + 1
+    )
+
+    var path = "${activity.getExternalFilesDirs("")[1]}"
+    val separator: String = activity.getString(R.string.android_directory_seperator)
+    val sepPos = path.indexOf(separator)
+    if (sepPos != -1) {
+      path = path.substring(0, sepPos)
+    }
```

Comment:
We are writing the same code twice. Utility or extension function would be great!